test/e2e/redundancy/gateway_failover.go

```diff
@@ -132,11 +132,20 @@ func testGatewayFailOverScenario(f *subFramework.Framework) {
 	gwPassive := f.AwaitGatewayWithStatus(framework.ClusterA, initialNonGatewayNode.Name, subv1.HAStatusPassive)
 	Expect(gwPassive.Status.Connections).To(BeEmpty(), "The passive gateway must have no connections")
 
+	// Start watching the API for Gateway deletions
+	gwInformer, stopInformer := f.GetGatewayInformer(framework.ClusterA)
+	defer close(stopInformer)
+
+	deleteCh := subFramework.GetDeletionChannel(gwInformer)
+
 	// Set the gateway label for the active gateway worker node to false so the submariner engine pod will be
 	// terminated.
 	By(fmt.Sprintf("Setting the gateway label for node %q to false", initialGatewayNode.Name))
 	f.SetGatewayLabelOnNode(framework.ClusterA, initialGatewayNode.Name, false)
 
+	By(fmt.Sprintf("Verifying that the gateway %q was deleted", initialGatewayNode.Name))
+	Eventually(deleteCh, 10.0).Should(Receive(Equal(initialGatewayNode.Name)))
```

Comment:
Perhaps we should use the same general timeout as `AwaitUntil`.

Reply:
:+1: